turn_alerts/tests/test_views.py

```diff
@@ -0,0 +1,384 @@
+                    "timestamp": "1735425598",
+                }
+            ]
+        }
+        response = self.client.post(
+            reverse("turn-messages", kwargs={"org_id": self.org.id + 1}),
+            data,
+            format="json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_org_does_not_exist(self):
+        """
+        Should return 400 if org does not exist
+        """
+
+        data = {
+            "statuses": [
+                {
+                    "errors": [
+                        {
+                            "code": 131026,
+                            "error_data": {"details": "Message Undeliverable."},
+                            "message": "Message undeliverable",
+                            "title": "Message undeliverable",
+                        }
+                    ],
+                    "id": "wamid.HBgFMzE1MzEVAgARGBIzMzQ2RUY0MUU4OTJGOEM5MjAA",
+                    "recipient_id": "31531",
+                    "status": "failed",
+                    "timestamp": "1735425598",
+                }
+            ]
+        }
+        response = self.client.post(
+            reverse("turn-messages", kwargs={"org_id": self.org.id + 1}),
+            data,
+            format="json",
+            HTTP_X_TURN_HOOK_SIGNATURE=self.generate_hmac_signature(data, "REPLACEME"),
+            HTTP_X_TURN_HOOK_SUBSCRIPTION="whatsapp",
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+
+class TestAlertsRoundTrip(APITestCase):
+    def setUp(self):
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+
+        self.org = create_org()
+        self.turnalerts_account = create_turnalerts(org=self.org)
+
+    def generate_hmac_signature(self, data, key):
+        data = JSONRenderer().render(data)
+        h = hmac.new(key.encode(), data, sha256)
+        return base64.b64encode(h.digest()).decode()
+
+    def test_delivery_failure_event_roundtrip(self):
+        """
+        Test roundtrip for status failed event
+        """
+        data = {
+            "statuses": [
+                {
+                    "errors": [
+                        {
+                            "code": 131026,
+                            "error_data": {"details": "Message Undeliverable."},
+                            "message": "Message undeliverable",
+                            "title": "Message undeliverable",
+                        }
+                    ],
+                    "id": "wamid.HBgFMzE1MzEVAgARGBIzMzQ2RUY0MUU4OTJGOEM5MjAA",
+                    "recipient_id": "2779631245",
+                    "status": "failed",
+                    "timestamp": "1735425598",
+                }
+            ]
+        }
+
+        url = reverse("turn-messages", kwargs={"org_id": self.org.id})
+
+        with patch("requests.post") as mock_post:
```

Comment:
Can we use responses to mock requests? or is this how we do it in the reset of sidekick?